checker/domain.go

```diff
@@ -124,25 +120,30 @@ func (d DomainQuery) lookupHostnames() ([]string, error) {
 //   `mxHostnames` is the list of expected hostnames.
 //     If `mxHostnames` is nil, we don't validate the DNS lookup.
 func CheckDomain(domain string, mxHostnames []string) DomainResult {
-	return performCheck(&DomainQuery{Domain: domain, ExpectedHostnames: mxHostnames})
+	return performCheck(DomainQuery{
+		Domain:            domain,
+		ExpectedHostnames: mxHostnames,
+		hostnameLookup:    &dnsLookup{},
+		hostnameChecker:   &tlsChecker{},
+	})
 }
 
-func performCheck(query domainCheckQuery) DomainResult {
+func performCheck(query DomainQuery) DomainResult {
 	result := DomainResult{
-		Domain:          query.getDomain(),
-		MxHostnames:     query.getExpectedHostnames(),
+		Domain:          query.Domain,
+		MxHostnames:     query.ExpectedHostnames,
 		HostnameResults: make(map[string]HostnameResult),
 	}
 	// 1. Look up hostnames
 	// 2. Perform and aggregate checks from those hostnames.
 	// 3. Set a summary message.
-	hostnames, err := query.lookupHostnames()
+	hostnames, err := query.hostnameLookup.lookup(query.Domain)
```

Comment:
Alternatively, you can take advantage of the anonymous field in the struct to pass the call straight through to lookup, eg. `query.lookup(query.Domain)`, (probably with a more descriptive name). Up to you!